state/endpoint_bindings.go

```diff
@@ -269,18 +298,18 @@ func validateEndpointBindingsForMachines(a *Application, oldBindings, newBinding
 		if err != nil {
 			return errors.Annotatef(err, "unable to get space assignments for machine %q", m.Id())
 		}
-		for spName := range machineSpaces {
-			machineCountInSpace[spName]++
+		for spID := range machineSpaces {
+			machineCountInSpace[spID]++
 		}
 	}
 
-	if newDefaultSpace, defined := newBindings[""]; defined && newDefaultSpace != network.DefaultSpaceName {
+	if newDefaultSpace, defined := newBindings[defaultEndpointName]; defined && newDefaultSpace != network.DefaultSpaceId {
```

Comment:
Actually this comparison is wrong (I have a fix for this in my PR). It should read:
```
defined && oldBindings[defaultEndpointName] != newDefaultSpace && newDefaultSpace != network.DefaultSpaceID
```
This means that we also need to bring back the oldBindings method argument but I will fix it in my PR so no need to worry about it.